metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/DataStructureFunctions.java

```diff
@@ -137,6 +138,105 @@ public Object apply(List<Object> args) {
     }
   }
 
+  @Stellar( namespace="HLLP"
+          , name="CARDINALITY"
+          , description="Returns HyperLogLogPlus-estimated cardinality for this set"
+          , params = { "hyperLogLogPlus - the hllp set" }
+          , returns = "Long value representing the cardinality for this set"
+  )
+  public static class HLLPCardinality extends BaseStellarFunction {
+
+    @Override
+    public Object apply(List<Object> args) {
+      if (args.size() < 1) {
+        throw new IllegalArgumentException("Must pass an hllp set to get the cardinality for");
+      }
+      return ((HyperLogLogPlus) args.get(0)).cardinality();
+    }
+  }
+
+  @Stellar( namespace="HLLP"
+          , name="INIT"
+          , description="Initializes the set"
+          , params = {
+                      "p (required) - the precision value for the normal set"
+                     ,"sp - the precision value for the sparse set. If sp is not specified the sparse set will be disabled."
+                     }
+          , returns = "A new HyperLogLogPlus set"
+  )
+  public static class HLLPInit extends BaseStellarFunction {
+
+    @Override
+    public Object apply(List<Object> args) {
+      if (args.size() == 0) {
+        throw new IllegalArgumentException("Normal set precision is required");
+      } else if (args.size() == 1) {
+        int p = ConversionUtils.convert(args.get(0), Integer.class);
+        return new HyperLogLogPlus(p);
+      } else {
+        int p = ConversionUtils.convert(args.get(0), Integer.class);
+        int sp = ConversionUtils.convert(args.get(1), Integer.class);
+        return new HyperLogLogPlus(p, sp);
+      }
+    }
+  }
+
+  @Stellar( namespace="HLLP"
+          , name="MERGE"
+          , description="Merge hllp sets together"
+          , params = {
+                      "hllp1 - first hllp set"
```

Comment:
Can we make this fit the pattern from `BLOOM_MERGE` to just take a list?  That was done because interactions with the profiler, which is largely where this will be used, will be made easier with that arrangement (i.e. `PROFILE_GET` returns a list of these, so you'd call this 99% of the time via `HLLP_MERGE(PROFILE_GET(....))`.